diff --git a/node_modules/lastfm-node-client/lib/ApiRequest.js b/node_modules/lastfm-node-client/lib/ApiRequest.js
index ecd39d0..c8e1cc5 100644
--- a/node_modules/lastfm-node-client/lib/ApiRequest.js
+++ b/node_modules/lastfm-node-client/lib/ApiRequest.js
@@ -91,7 +91,7 @@ class ApiRequest {
 		const paramsStr = querystring.stringify(paramsObj);
 		const options = {
 			hostname: "ws.audioscrobbler.com",
-			path: "/2.0"
+			path: "/2.0",
 		};
 
 		if (method === "POST") {
@@ -113,7 +113,11 @@ class ApiRequest {
 				httpResponse.on("data", chunk => data += chunk);
 				httpResponse.on("end", () => resolve(data));
 				httpResponse.on("error", err => reject(err));
-			});
+			})
+				// stop waiting for request if it takes longer than 3 seconds
+				.setTimeout(3000, () => {
+					httpRequest.destroy(new Error('ETIMEDOUT - socket hang up'));
+				});
 
 			httpRequest.on("error", err => reject(err));
 
diff --git a/node_modules/lastfm-node-client/lib/LastFm.js b/node_modules/lastfm-node-client/lib/LastFm.js
index 2393a4d..75bdf37 100644
--- a/node_modules/lastfm-node-client/lib/LastFm.js
+++ b/node_modules/lastfm-node-client/lib/LastFm.js
@@ -1045,7 +1045,6 @@ class LastFm {
 	 * @param {callback} [callback]
 	 * @returns {(Promise|LastFm)}
 	 */
-
 	trackUpdateNowPlaying(params, callback) {
 		const apiRequest = new ApiRequest()
 			.set(params)
