diff --git a/node_modules/lastfm-ts-api/dist/api-request.js b/node_modules/lastfm-ts-api/dist/api-request.js
index d3d3287..20509af 100644
--- a/node_modules/lastfm-ts-api/dist/api-request.js
+++ b/node_modules/lastfm-ts-api/dist/api-request.js
@@ -64,7 +64,7 @@ export class LastFMApiRequest {
             }
             httpRequest.end();
         }).then(([response, content]) => {
-            if (response.headers['content-type'] !== 'application/json') {
+            if (!response.headers['content-type']?.includes('application/json')) {
                 throw new LastFMResponseError(`lastfm-ts-api: Expected JSON response but received '${response.headers['content-type']}'`, { response, content });
             }
             let data;
