diff --git a/node_modules/musicbrainz-api/lib/http-client.js b/node_modules/musicbrainz-api/lib/http-client.js
index a5ca5f7..04b9ef8 100644
--- a/node_modules/musicbrainz-api/lib/http-client.js
+++ b/node_modules/musicbrainz-api/lib/http-client.js
@@ -9,6 +9,7 @@ function isConnectionReset(err) {
 export class HttpClient {
     constructor(httpOptions) {
         this.httpOptions = httpOptions;
+        this.preRequest = (httpOptions ?? {}).preRequest
     }
     get(path, options) {
         return this._fetch('get', path, options);
@@ -36,6 +37,9 @@ export class HttpClient {
         if (cookies !== null) {
             headers.set('Cookie', cookies);
         }
+        if(this.preRequest !== undefined) {
+            this.preRequest(method, url, headers);
+        }
         while (retryLimit > 0) {
             let response;
             try {
diff --git a/node_modules/musicbrainz-api/lib/musicbrainz-api-node.js b/node_modules/musicbrainz-api/lib/musicbrainz-api-node.js
index c9cad5a..e2cf725 100644
--- a/node_modules/musicbrainz-api/lib/musicbrainz-api-node.js
+++ b/node_modules/musicbrainz-api/lib/musicbrainz-api-node.js
@@ -15,7 +15,8 @@ export class MusicBrainzApi extends MusicBrainzApiDefault {
         return new HttpClientNode({
             baseUrl: this.config.baseUrl,
             timeout: 500,
-            userAgent: `${this.config.appName}/${this.config.appVersion} ( ${this.config.appContactInfo} )`
+            userAgent: `${this.config.appName}/${this.config.appVersion} ( ${this.config.appContactInfo} )`,
+            preRequest: this.config.preRequest
         });
     }
     async login() {
diff --git a/node_modules/musicbrainz-api/lib/musicbrainz-api.d.ts b/node_modules/musicbrainz-api/lib/musicbrainz-api.d.ts
index cd05017..684c7cf 100644
--- a/node_modules/musicbrainz-api/lib/musicbrainz-api.d.ts
+++ b/node_modules/musicbrainz-api/lib/musicbrainz-api.d.ts
@@ -2,6 +2,7 @@ import type { XmlMetadata } from './xml/xml-metadata.js';
 import { RateLimitThreshold } from 'rate-limit-threshold';
 import * as mb from './musicbrainz.types.js';
 import { HttpClient, type MultiQueryFormData } from "./http-client.js";
+import { Url } from 'node:url';
 export { XmlMetadata } from './xml/xml-metadata.js';
 export { XmlIsrc } from './xml/xml-isrc.js';
 export { XmlIsrcList } from './xml/xml-isrc-list.js';
@@ -75,6 +76,7 @@ export interface IMusicBrainzConfig {
      * Default is [15, 18], which allows up to 15 requests every 18 seconds
      */
     rateLimit?: [number, number];
+    preRequest?: (method: string, url: string, headers: Headers) => void
 }
 interface IInternalConfig extends IMusicBrainzConfig {
     baseUrl: string;
diff --git a/node_modules/musicbrainz-api/lib/musicbrainz-api.js b/node_modules/musicbrainz-api/lib/musicbrainz-api.js
index 72fb15d..a860223 100644
--- a/node_modules/musicbrainz-api/lib/musicbrainz-api.js
+++ b/node_modules/musicbrainz-api/lib/musicbrainz-api.js
@@ -43,7 +43,8 @@ export class MusicBrainzApi {
         return new HttpClient({
             baseUrl: this.config.baseUrl,
             timeout: 500,
-            userAgent: `${this.config.appName}/${this.config.appVersion} ( ${this.config.appContactInfo} )`
+            userAgent: `${this.config.appName}/${this.config.appVersion} ( ${this.config.appContactInfo} )`,
+            preRequest: this.config.preRequest
         });
     }
     async restGet(relUrl, query = {}) {
