diff --git a/node_modules/musicbrainz-api/lib/musicbrainz-api.d.ts b/node_modules/musicbrainz-api/lib/musicbrainz-api.d.ts
index dfd2937..09ed531 100644
--- a/node_modules/musicbrainz-api/lib/musicbrainz-api.d.ts
+++ b/node_modules/musicbrainz-api/lib/musicbrainz-api.d.ts
@@ -64,6 +64,7 @@ export interface IMusicBrainzConfig {
      */
     appContactInfo?: string;
     disableRateLimiting?: boolean;
+    rateLimit?: [number, number]
 }
 interface IInternalConfig extends IMusicBrainzConfig {
     baseUrl: string;
diff --git a/node_modules/musicbrainz-api/lib/musicbrainz-api.js b/node_modules/musicbrainz-api/lib/musicbrainz-api.js
index 299be1c..7ee41b0 100644
--- a/node_modules/musicbrainz-api/lib/musicbrainz-api.js
+++ b/node_modules/musicbrainz-api/lib/musicbrainz-api.js
@@ -36,7 +36,8 @@ export class MusicBrainzApi {
             ..._config
         };
         this.httpClient = this.initHttpClient();
-        this.rateLimiter = new RateLimitThreshold(15, 18);
+        const limits = this.config.rateLimit ?? [15,18];
+        this.rateLimiter = new RateLimitThreshold(limits[0], limits[1]);
     }
     initHttpClient() {
         return new HttpClient({
