diff --git a/node_modules/winston-daily-rotate-file/index.js b/node_modules/winston-daily-rotate-file/index.js
index c818660..a8d3040 100644
--- a/node_modules/winston-daily-rotate-file/index.js
+++ b/node_modules/winston-daily-rotate-file/index.js
@@ -1,6 +1,6 @@
 'use strict';
 
-var winston = require('winston');
+var winston = require('@foxxmd/winston');
 var DailyRotateFile = require('./daily-rotate-file');
 
 winston.transports.DailyRotateFile = DailyRotateFile;
